lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/client/YarnSubmissionHelper.java

```diff
@@ -241,6 +242,29 @@ public YarnSubmissionHelper setConfigurationFilePaths(final List<String> configu
   }
 
   /**
+   * Sets environment variable map.
+   * @param map
+   * @return
+   */
+  public YarnSubmissionHelper setJobSubmissionEnvMap(final Map<String, String> map) {
```

Comment:
This API is unclear: Do I set the whole map, or one entry at a time?

Reply:
Both set entire map and add individual entries are supported.